Kuroba/app/src/main/java/com/github/adamantcheese/chan/core/manager/WatchManager.java

```diff
@@ -99,28 +105,43 @@
 
     private final DatabaseManager databaseManager;
     private final DatabasePinManager databasePinManager;
+    private final DatabaseSavedThreadManager databaseSavedThreadManager;
     private final ChanLoaderFactory chanLoaderFactory;
     private final WakeManager wakeManager;
     private final PageRequestManager pageRequestManager;
+    private final ThreadSaveManager threadSaveManager;
 
     private IntervalType currentInterval = IntervalType.NONE;
 
     private final List<Pin> pins;
+    private final List<SavedThread> savedThreads;
+    private boolean prevIncrementalThreadSavingEnabled = false;
```

Comment:
Initialize it based on the isWatcherEnabled && isBackgroundWatcherEnabled flags instead of always initializing it with false.